packages/request-idle-callback-polyfill/npm/index.js

```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+// @TODO figure out if we need a prod/dev build for this?
+// if (process.env.NODE_ENV === 'production') {
```

Comment:
Yes :D I'm curious why you thought perhaps we don't need separate builds?

Reply:
I see. Yeah, let's have both, for debugging. And since it's possible we may add warnings.

Reply:
This may limit adoption and someone might publish one that just has the one. The prod/dev build thing is still somewhat controversial.

The whole point of this is that we would like this particular implementation to be canonical and gain adoption so a complex build works against that.

Reply:
Why do we care about anyone who isn’t also using React, with which they have to deal with this anyway?

Reply:
I get that you’re saying we can get more adoption within the React community if it becomes widely adopted outside of it but idk if I buy that scenario

Reply:
In my view the main argument for a DEV build is so we can warn, e.g. if we detect the presence of a non-native global requestIdleCallback implementation, which isn’t something we’d want to do in prod (I’m assuming)

Reply:
Polyfills are not always part of the same build system neither since they can often just be concatenated or loaded from a polyfill CDN service. Even if you're using React you'd hit this.

I just don't see a reason for it.

The scenario you're describing is just a warning about two polyfills existing which you shouldn't do anyway. Even if we warn, there might be another one after that overrides.

We can still warn for a non-native global requestIdleCallback implementation in React itself. That's where it matters.

Reply:
Warning in ReactDOM makes sense.